src/Cli/dotnet/commands/dotnet-workload/install/FileBasedInstaller.cs

```diff
@@ -348,6 +365,29 @@ public void GarbageCollectInstalledWorkloadPacks(DirectoryPath? offlineCache = n
                     Directory.Delete(packIdDir);
                 }
             }
+
+            if (cleanAllPacks)
+            {
+                EradicateAllWorkloadInstallationRecords(_sdkFeatureBand);
+            }
+        }
+
+        public void EradicateAllWorkloadInstallationRecords(SdkFeatureBand featureBand)
```

Comment:
Suggest renaming this variable to avoid making it seem like we're only deleting the workload installation records for the specified band.

```suggestion
        public void EradicateAllWorkloadInstallationRecords(SdkFeatureBand currentFeatureBand)
```